TST: groupstars tests too senstitive?

I encountered the following failures when testing `photutils` locally on Windows 10.
```
============================= test session starts =============================
platform win32 -- Python 3.7.1, pytest-3.10.0, py-1.7.0, pluggy-0.8.0

Running tests with photutils version 0.7.dev3307.
Running tests in photutils docs.

Date: 2019-01-06T02:10:26

Platform: Windows-10-10.0.17134-SP0

Executable: C:\...\Miniconda3\envs\py37\python.exe

Full Python Version:
3.7.1 (default, Oct 28 2018, 08:39:03) [MSC v.1912 64 bit (AMD64)]

encodings: sys: utf-8, locale: cp1252, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.15.4
Scipy: 1.1.0
Matplotlib: 2.2.3
Cython: 0.29
Astropy: 3.1.1
scikit-image: 0.14.1
scikit-learn: 0.20.0
astropy_helpers: 3.1
Using Astropy options: remote_data: none.

Matplotlib: 2.2.3
Freetype: 2.9.1
rootdir: ..., inifile: setup.cfg
plugins: xdist-1.24.0, remotedata-0.3.1, openfiles-0.3.0, mpl-0.10, forked-0.2,doctestplus-0.1.3, arraydiff-0.2

...

================================== FAILURES ===================================
____________________ TestDBSCANGroup.test_group_stars_one _____________________

object = <photutils.psf.tests.test_groupstars.TestDBSCANGroup object at 0x0000023B06CEB2B0>

    def test_group_stars_one(object):
        x_0 = np.array([0, np.sqrt(2)/4, np.sqrt(2)/4, -np.sqrt(2)/4,
                        -np.sqrt(2)/4])
        y_0 = np.array([0, np.sqrt(2)/4, -np.sqrt(2)/4, np.sqrt(2)/4,
                        -np.sqrt(2)/4])
        x_1 = x_0 + 2.0
        first_group = Table([x_0, y_0, np.arange(len(x_0)) + 1,
                            np.ones(len(x_0), dtype=np.int)],
                            names=('x_0', 'y_0', 'id', 'group_id'))
        second_group = Table([x_1, y_0, len(x_0) + np.arange(len(x_0)) + 1,
                              2*np.ones(len(x_0), dtype=np.int)],
                             names=('x_0', 'y_0', 'id', 'group_id'))
        starlist = vstack([first_group, second_group])
        dbscan = DBSCANGroup(crit_separation=0.6)
        test_starlist = dbscan(starlist['x_0', 'y_0', 'id'])
>       assert_array_equal(starlist, test_starlist)
E       AssertionError:
E       Arrays are not equal
E
E       (mismatch 100.0%)
E        x: array([( 0.      ,  0.      ,  1, 1), ( 0.353553,  0.353553,  2, 1),
E              ( 0.353553, -0.353553,  3, 1), (-0.353553,  0.353553,  4, 1),
E              (-0.353553, -0.353553,  5, 1), ( 2.      ,  0.      ,  6, 2),...
E        y: array([( 0.      ,  0.      ,  1, 1), ( 0.353553,  0.353553,  2, 1),
E              ( 0.353553, -0.353553,  3, 1), (-0.353553,  0.353553,  4, 1),
E              (-0.353553, -0.353553,  5, 1), ( 2.      ,  0.      ,  6, 2),...

photutils\psf\tests\test_groupstars.py:289: AssertionError
____________________ TestDBSCANGroup.test_group_stars_two _____________________

object = <photutils.psf.tests.test_groupstars.TestDBSCANGroup object at 0x0000023B045D73C8>

    def test_group_stars_two(object):
        first_group = Table([1.5*np.ones(5), np.linspace(0, 1, 5),
                             np.arange(5) + 1, np.ones(5, dtype=np.int)],
                            names=('x_0', 'y_0', 'id', 'group_id'))
        second_group = Table([1.5*np.ones(5), np.linspace(2, 3, 5),
                              6 + np.arange(5), 2*np.ones(5, dtype=np.int)],
                             names=('x_0', 'y_0', 'id', 'group_id'))
        third_group = Table([np.linspace(0, 1, 5), 1.5*np.ones(5),
                             11 + np.arange(5), 3*np.ones(5, dtype=np.int)],
                            names=('x_0', 'y_0', 'id', 'group_id'))
        fourth_group = Table([np.linspace(2, 3, 5), 1.5*np.ones(5),
                              16 + np.arange(5), 4*np.ones(5, dtype=np.int)],
                             names=('x_0', 'y_0', 'id', 'group_id'))
        starlist = vstack([first_group, second_group, third_group,
                           fourth_group])
        dbscan = DBSCANGroup(crit_separation=0.3)
        test_starlist = dbscan(starlist['x_0', 'y_0', 'id'])
>       assert_array_equal(starlist, test_starlist)
E       AssertionError:
E       Arrays are not equal
E
E       (mismatch 100.0%)
E        x: array([(1.5 , 0.  ,  1, 1), (1.5 , 0.25,  2, 1), (1.5 , 0.5 ,  3, 1),
E              (1.5 , 0.75,  4, 1), (1.5 , 1.  ,  5, 1), (1.5 , 2.  ,  6, 2),
E              (1.5 , 2.25,  7, 2), (1.5 , 2.5 ,  8, 2), (1.5 , 2.75,  9, 2),...
E        y: array([(1.5 , 0.  ,  1, 1), (1.5 , 0.25,  2, 1), (1.5 , 0.5 ,  3, 1),
E              (1.5 , 0.75,  4, 1), (1.5 , 1.  ,  5, 1), (1.5 , 2.  ,  6, 2),
E              (1.5 , 2.25,  7, 2), (1.5 , 2.5 ,  8, 2), (1.5 , 2.75,  9, 2),...

photutils\psf\tests\test_groupstars.py:308: AssertionError
____________________ TestDBSCANGroup.test_isolated_sources ____________________

self = <photutils.psf.tests.test_groupstars.TestDBSCANGroup object at 0x0000023B045D7860>

    def test_isolated_sources(self):
        """
        Test case when all sources are isolated.
        """
        x_0 = np.array([0, np.sqrt(2)/4, np.sqrt(2)/4, -np.sqrt(2)/4,
                        -np.sqrt(2)/4])
        y_0 = np.array([0, np.sqrt(2)/4, -np.sqrt(2)/4, np.sqrt(2)/4,
                        -np.sqrt(2)/4])
        starlist = Table([x_0, y_0, np.arange(len(x_0)) + 1,
                          np.arange(len(x_0)) + 1],
                         names=('x_0', 'y_0', 'id', 'group_id'))
        dbscan = DBSCANGroup(crit_separation=0.01)
        test_starlist = dbscan(starlist['x_0', 'y_0', 'id'])
>       assert_array_equal(starlist, test_starlist)
E       AssertionError:
E       Arrays are not equal
E
E       (mismatch 100.0%)
E        x: array([( 0.      ,  0.      , 1, 1), ( 0.353553,  0.353553, 2, 2),
E              ( 0.353553, -0.353553, 3, 3), (-0.353553,  0.353553, 4, 4),
E              (-0.353553, -0.353553, 5, 5)],
E             dtype=[('x_0', '<f8'), ('y_0', '<f8'), ('id', '<i4'), ('group_id', '<i4')])
E        y: array([( 0.      ,  0.      , 1, 1), ( 0.353553,  0.353553, 2, 2),
E              ( 0.353553, -0.353553, 3, 3), (-0.353553,  0.353553, 4, 4),
E              (-0.353553, -0.353553, 5, 5)],
E             dtype=[('x_0', '<f8'), ('y_0', '<f8'), ('id', '<i4'), ('group_id', '<i8')])

photutils\psf\tests\test_groupstars.py:323: AssertionError
_______________________ TestDBSCANGroup.test_id_column ________________________

self = <photutils.psf.tests.test_groupstars.TestDBSCANGroup object at 0x0000023B04597FD0>

    def test_id_column(self):
        x_0 = np.array([0, np.sqrt(2)/4, np.sqrt(2)/4, -np.sqrt(2)/4,
                        -np.sqrt(2)/4])
        y_0 = np.array([0, np.sqrt(2)/4, -np.sqrt(2)/4, np.sqrt(2)/4,
                        -np.sqrt(2)/4])
        starlist = Table([x_0, y_0, np.arange(len(x_0)) + 1,
                          np.arange(len(x_0)) + 1],
                         names=('x_0', 'y_0', 'id', 'group_id'))
        dbscan = DBSCANGroup(crit_separation=0.01)
        test_starlist = dbscan(starlist['x_0', 'y_0'])
>       assert_array_equal(starlist, test_starlist)
E       AssertionError:
E       Arrays are not equal
E
E       (mismatch 100.0%)
E        x: array([( 0.      ,  0.      , 1, 1), ( 0.353553,  0.353553, 2, 2),
E              ( 0.353553, -0.353553, 3, 3), (-0.353553,  0.353553, 4, 4),
E              (-0.353553, -0.353553, 5, 5)],
E             dtype=[('x_0', '<f8'), ('y_0', '<f8'), ('id', '<i4'), ('group_id', '<i4')])
E        y: array([( 0.      ,  0.      , 1, 1), ( 0.353553,  0.353553, 2, 2),
E              ( 0.353553, -0.353553, 3, 3), (-0.353553,  0.353553, 4, 4),
E              (-0.353553, -0.353553, 5, 5)],
E             dtype=[('x_0', '<f8'), ('y_0', '<f8'), ('id', '<i4'), ('group_id', '<i8')])

photutils\psf\tests\test_groupstars.py:335: AssertionError
```

--- a/photutils/geometry/circular_overlap.pyx
+++ b/photutils/geometry/circular_overlap.pyx
@@ -1,4 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+#cython: language_level=3
 """
 The functions defined here allow one to determine the exact area of
 overlap of a rectangle and a circle (written by Thomas Robitaille).

--- a/photutils/geometry/core.pxd
+++ b/photutils/geometry/core.pxd
@@ -1,4 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+#cython: language_level=3
 
 # This file is needed in order to be able to cimport functions into other Cython files
 

--- a/photutils/geometry/core.pyx
+++ b/photutils/geometry/core.pyx
@@ -1,4 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+#cython: language_level=3
 """
 The functions here are the core geometry functions.
 """

--- a/photutils/geometry/elliptical_overlap.pyx
+++ b/photutils/geometry/elliptical_overlap.pyx
@@ -1,4 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+#cython: language_level=3
 """
 The functions defined here allow one to determine the exact area of
 overlap of an ellipse and a triangle (written by Thomas Robitaille).

--- a/photutils/geometry/rectangular_overlap.pyx
+++ b/photutils/geometry/rectangular_overlap.pyx
@@ -1,4 +1,5 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
+#cython: language_level=3
 
 import numpy as np
 cimport numpy as np

